cmd/eib/main.go

```diff
@@ -24,43 +25,176 @@ import (
 )
 
 const (
-	argConfigFile = "config-file"
-	argConfigDir  = "config-dir"
-	argBuildDir   = "build-dir"
-	argValidate   = "validate"
+	argDefinitionFile = "config-file"
+	argConfigDir      = "config-dir"
+	argBuildDir       = "build-dir"
+	argValidate       = "validate"
 )
 
-func processArgs() (*image.Context, error) {
-	var (
-		configFile   string
-		configDir    string
-		rootBuildDir string
-		validate     bool
-	)
-
-	flag.StringVar(&configFile, argConfigFile, "", "name of the image configuration file")
-	flag.StringVar(&configDir, argConfigDir, "", "full path to the image configuration directory")
-	flag.StringVar(&rootBuildDir, argBuildDir, "", "full path to the directory to store build artifacts")
-	flag.BoolVar(&validate, argValidate, false, "if specified, the image definition will be validated but not built")
+const (
+	exitCodeDefinitionParse   = 10
+	exitCodeDefinitionInvalid = 11
+	exitCodeConfigDir         = 12
+	exitCodeBuildDir          = 13
+	exitCodeRpmDependencies   = 14
```

Comment:
I'm not particularly against these but I want to raise the question of how do we track / document custom exit status codes. AFAIK there is some kind of standartisation in place but I'm not familiar enough with it. Please ignore if our lovely PM (@rdoxenham) is fine with using these.

Reply:
Admittedly, I don't have a ton of experience with this in the past. I found myself exiting with 1 a lot and it felt like I should at least take a shot doing it less wrong.

Reply:
There are quite a few commonly-used exit codes in Linux and I don't suggest that we re-use these. See here: https://tldp.org/LDP/abs/html/exitcodes.html

Reply:
I intentionally didn't use any in that table. Thankfully, the range is available codes is pretty large.

Reply:
Yeah, they are, but if you look in other sources it seems that exit codes under 133 are also widely used, so that link I shared isn't exhaustive. Maybe we can go with 166 and above just to be on the safe side?

Reply:
I'm also completely fine with just using 1 for all error exists and then revisiting on the off chance anyone ever asks us for this. That's probably the safest approach to start with, and I'd rather add complexity when it's actually needed/going to be used.